### ISSUE TITLE
Add option to let the container stop gracefully after docker-entrypoint-initdb.d

### ISSUE THREAD

**Issue by metas-ts** [docker-library/postgres#424]
Hi,
I have two usage scenarios where imho it would make a lot of sense to have the option to just init a postgres DBMS via scripts in `/docker-entrypoint-initdb.d` and then have the docker container exit gracefully as soon as those init scripts are run.

This would support my following use cases:
* on my jenkins, I would like to get an initial pg-dump, then apply some migration scripts and then create and publish a new pg-dump. As soon as that's done, the docker container should stop.
* in my kubernetes cluster, I would like to use a postgres docker container as a [kubernetes init-container](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/) to prepare the grounds before the *real* postgresql container actually starts. Why: the init-container's image contains a lot of stuff that is only needed for database inititalization (in my case  curl and a java runtime environment...yeah, better don't ask ;-) )..but it writes the database files into a volume which is then used by the *real* production postgres container.

wdyt?

**Comment by eric-cancil** [comment]
+1 - this would be super useful for seeding data into a container and then recommitting it (think allowing developers to pull a container with seeded data already inside it

**Comment by GiannisSialmas** [comment]
A hacky way i found to accomplish not starting the database after populating it, is by using a multi-stage build. 

In the first stage i remove the last line from `/docker-entrypoint.sh` which seems to be responsible for starting the server. I do in the Dockerfile with a **RUN** command and **sed**. I then trigger the `/docker-entrypoint.sh` script manually to populate my database. 

In the second stage i use the same postgres image. I copy the database directory from my first build but this time i leave `/docker-entrypoint.sh` so that it can start the database when the container is started. 

You can check out my Dockerfile which i use to do a build of a populated chinook database [here](https://github.com/GiannisSialmas/docker/blob/master/postgres/chinook/Dockerfile).

**Comment by GiannisSialmas** [comment]
The last line of the `docker-entrypoint.sh` seems to be responsible for starting the database.That is
```
exec "$@"
```
As a hacky way to accomplish not starting the database, i currently remove the last line of `docker-entrypoint.sh` using a `RUN` command with sed inside the Dockerfile like this:
```
RUN sed -i '$ d' /usr/local/bin/docker-entrypoint.sh
``` 
As a more permanent solution, an ENV variable (e.g $DONT_START_DATABASE) and a change to the last lines of the `docker-entrypoint.sh` to exit the script succesfully before starting the database would be great. Example:
```
if [ "$DONT_START_DATABASE" = true ]; then
	exit 0;
fi
exec "$@"
```
As a plus, other people using the hacky solution with sed wouldn't be affected by the change.

**Comment by gsf** [comment]
I have a similar kubernetes situation with the following in an init container:

```yaml
- name: setup
  image: postgres
  args:
  # Trick entrypoint into exiting after setup
  - postgres
  - --version
```

**Comment by tianon** [comment]
I think the `--version` or `sed`-based solution (https://github.com/docker-library/postgres/issues/424#issuecomment-445552662) is the best we can offer right now, but https://github.com/docker-library/postgres/pull/496 should make this process completely customizable (which should make this a lot easier to accomplish).

**Comment by tianon** [comment]
This is now possible to accomplish properly/cleanly via #496. :tada: :+1:

**Comment by Akkarine** [comment]
> I have a similar kubernetes situation with the following in an init container:
> 
> ```yaml
> - name: setup
>   image: postgres
>   args:
>   # Trick entrypoint into exiting after setup
>   - postgres
>   - --version
> ```

Looks like trick does not work anymore, as now check for command happens: https://github.com/docker-library/postgres/blob/b80fcb5ac7f6dde712e70d2d53a88bf880700fde/12/docker-entrypoint.sh#L275

**Comment by luckydonald** [comment]
@tianon It's a bit unclear now, how exactly can this be done with the landed #496?

**Comment by tianon** [comment]
We added an explicit example in https://github.com/docker-library/postgres/pull/1150, which happens to solve this specific use case.
